### ISSUE TITLE
Status bar tabs don't work anymore

### ISSUE THREAD

**Issue by olmokramer** [atom-community/linter#754]
I can't change between tabs anymore. I'm stuck on the file tab, and when I click the project tab nothing happens.

**Comment by steelbrain** [comment]
@atom-community/linter-contributors can someone else reproduce this?

**Comment by olmokramer** [comment]
By the way, the versions I'm using:

Atom 1.0.3-539b2b4 on Linux
Linter 1.2.4

**Comment by steelbrain** [comment]
Super-weird, can you try `apm install linter@1.2.3` and then downgrade one by one and pint point a version where this bug was introduced?

**Comment by olmokramer** [comment]
Sure I'll try :)

Another thing with the tabs, I tried enabling the line tab to see if that was working - it isn't - but I have to reload Atom before the line tab becomes visible. Should I open another issue for that?

**Comment by olmokramer** [comment]
Ok so I had to go back to `v1.1.0` for the tabs to work again.

**Comment by olmokramer** [comment]
I found the issue. It's my own [move-status-items](https://atom.io/packages/move-status-items) package, that cancelled the click. I'm sorry to have bothered you :)

**Comment by steelbrain** [comment]
Glad you found the problem :)

**Comment by olmokramer** [comment]
Ok, I found the real issue. It's not the click event that's not triggered, but the emitter got disposed because I call `statusBarLeftElement.appendChilc(linterBottomContainer)` in `move-status-items` and apparently that detaches the element. Would it be possible to not dispose everything in the `detachedCallback` of `lib/views/bottom-container` but instead move it into a destroy method that gets called when the element is truly destroyed?

**Comment by steelbrain** [comment]
@olmokramer I didn't know there's a `destroyedCallback` for HTML5 custome elements that gets called when the element is destroyed, at least I can't find it on http://www.html5rocks.com/en/tutorials/webcomponents/customelements/

**Comment by olmokramer** [comment]
No, there isnt. I meant a custom method, which you can call at the point the tab container is destroyed. For example, right after `@bottomBar.destroy()` on [line 135](https://github.com/atom-community/linter/blob/master/lib/linter-views.coffee#L135) of `lib/linter-views.coffee` also call `@bottomContainer.destroy()` which will then dispose of the emitter of the bottomContainer.

**Comment by steelbrain** [comment]
@olmokramer I already have it planned :)

**Comment by olmokramer** [comment]
Great, thanks!

**Comment by olmokramer** [comment]
@steelbrain Are you still planning on doing this? I still can't use the tabs in the status bar :cry:

**Comment by steelbrain** [comment]
@olmokramer I thought you fixed it. I am on Atom Slack if you wanna talk.

**Comment by olmokramer** [comment]
Hi, sorry for the late reply. No, there's nothing I can do to fix it. I can't reorder the items in the status bar without triggering the `detachedCallback` of the elements in the status bar, because even the following triggers the `deatchedCallback`:

```js
var div = document.createElement('div');
var custom = document.createElement('my-element');
div.appendChild(custom);

// this triggers the detachedCallback of my-element
div.appendChild(custom);
```

But `linter-bottom-container` disposes of its event listeners in the `detachedCallback`, so after the `statusBarLeftContainer.appendChild(linterBottomContainer)` the tabs stop working.

I suggested you use a method for the disposal of the event listeners that you call explicitly, something like a `destroy` method. I can do a PR if you want?

**Comment by steelbrain** [comment]
I would love to review a pr that makes them disposable.

**Comment by steelbrain** [comment]
Just so you would know, I've made them disposable with #939

**Comment by olmokramer** [comment]
Thanks! I'll check it out right now

**Comment by olmokramer** [comment]
Yep, that works great :fireworks: Thanks!
